### DISCUSSION TITLE
How do I use the Halide::Adams2019 AutoScheduler from CMake?

### DISCUSSION THREAD

**Discussion by l-oneil** [halide/Halide#5258] [Q&A] [answered]
Hi,

Just built a clean checkout of the latest branch, with the hope to try out the `Halide::Adams2019` scheduler.

I saw in the CMake documentation [that I should be able to do this] 
(https://github.com/halide/Halide/blame/master/README_cmake.md#L648) in order to utilise it. 

But when I try this, I get the following build error: 
`Error: Unknown autoscheduler name 'Halide::Adams2019'; known names are:`

Can anyone point out what I might be doing wrong please? 

Kind  Regards,
Liam

**Comment by alexreinking** [comment, marked as answer]
Hi @l-oneil -- the autoschedulers are being reworked a bit in #5228 so they aren't being released at the moment. We're hoping to have this fixed ASAP, ideally by next Friday.

**Reply by l-oneil** [reply]
Thanks for letting me know! Looking forward to trying them out :)

**Reply by alexreinking** [reply]
Of course! I'll try to keep you posted, but also subscribe to that PR so you know when it's merged 🙂

**Reply by l-oneil** [reply]
Hey @alexreinking, I've pulled the most recent version of Halide now (tagged v10.0.0), but I am still being hit with `Error: Unknown autoscheduler name 'Halide::Adams2019'; known names are:` when attempting to compile my generator. 

I see the update log is referring to the schedulers as being treated as "plugins" is there a build step or compile flag I am missing on top of the typical Halide CMake compilation process?

**Reply by alexreinking** [reply]
How are you compiling your generator? It should look something like:
```
add_halide_library(my_lib FROM my gen
                   AUTOSCHEDULER Halide::Adams2019)
```

**Reply by l-oneil** [reply]
Thanks for the quick reply!

Yeah sure, that's exactly the function call I am using, the error response I am getting says that the only known autoscheduler name is `Mullapudi2016`.

So just to confirm, I am using the following CMake command to compile v10.0.0:
` cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_DIR=/path/to/llvm-install/lib/cmake/llvm /path/to/Halide`

Is there anything extra I should be flagging to build Halide for access to `Adams2019`? I see that it's susposed to be used as a plugin? I'm not sure what that means exactly?

**Reply by alexreinking** [reply]
Hmm, no, Halide should build Adams2019 unless you explicitly set BUILD_SHARED_LIBS=NO

**Reply by l-oneil** [reply]
hmm, OK, this is making me feel like I have very likely done something stupid with my system paths. 
My bad, I'll try a clean install. 

Thanks for confirming :)
